pkg/process/procutil/data_scrubber_windows.go

```diff
@@ -20,3 +25,66 @@ var (
 	// it's for handling parameters like `/p` and `/rp` in windows
 	forbiddenSymbolsRegex = "[^/a-zA-Z0-9_*]"
 )
+
+var executibleExtensions = []string{".com", ".exe", ".bat", ".cmd", ".vbs", ".vbe", ".js", ".jse", ".wsf", ".wsh", ".psc1", ".ps1"}
+
+// stripArguments identifies windows extension and extracts command. Otherwise, returns the first element of the cmdline before first space.
+// If cmdline is empty, stripArguments will return an empty string.
+func (ds *DataScrubber) stripArguments(cmdline []string) []string {
+	if len(cmdline) < 1 {
+		return cmdline
+	}
+
+	strCmdline := cmdline[0]
+
+	// Case 1: OS has already completed splitting as there is one token per element, we return first token.
+	if len(cmdline) > 1 {
+		return []string{strCmdline}
+	}
+
+	// Case 2: One string for cmdline, use extensionParser() to find first token.
+	if len(cmdline) == 1 && !strings.HasPrefix(strCmdline, "\"") {
+		strippedCmdline := extensionParser(strCmdline, executibleExtensions)
+
+		if strippedCmdline != "" {
+			return []string{strings.TrimSuffix(strippedCmdline, " ")}
+		}
+
+		// If no extension is found, return first token of cmdline.
+		return []string{strings.TrimSuffix(strings.SplitN(strCmdline, " ", 2)[0], " ")}
+	}
+
+	// Case 2b: One string for cmdline and first token wrapped in quotes, use findEmbeddedQuotes() to find content between quotes.
+	strippedCmdline := findEmbeddedQuotes(strCmdline)
+	return []string{strings.TrimSuffix(strippedCmdline, " ")}
+}
+
+// extensionParser returns substring of cmdline up to the first extension (inclusive).
+// If no extension is found, returns empty string.
+// Example: Input="C:\\Program Files\\Datadog\\agent.vbe check process"  Output="C:\\Program Files\\Datadog\\agent.vbe"
+func extensionParser(cmdline string, executibleExtensions []string) string {
+	for _, c := range executibleExtensions {
+		// If extension is found before a word break (space or end of line).
+		if i := strings.Index(cmdline, c); i != -1 && (i+len(c) == len(cmdline) || cmdline[i+len(c)] == ' ') {
+			processedCmdline := cmdline[:i+len(c)]
+			return processedCmdline
+		}
+	}
+	return ""
+}
+
+// findEmbeddedQuotes returns the content between the first pair of double quotes in cmdline.
+// If there is no pair of double quotes found, function returns original cmdline.
+// Example: Input="\"C:\\Program Files\\Datadog\\agent.vbe\" check process" Output="C:\\Program Files\\Datadog\\agent.vbe"
+func findEmbeddedQuotes(cmdline string) string {
+	if len(cmdline) == 0 {
```

Comment:
This is not possible, right? Since this function assumes the first character is a `"`

Reply:
You are correct, this shouldn't be possible - I also just realized this bounds check shouldn't be necessary since there is already a length check after the split.